callback_plugins/custom_logger.py

```diff
@@ -1,72 +1,69 @@
-from __future__ import (absolute_import, division, print_function)
-#__metaclass__ = type
-
-import os
-
 from ansible.plugins.callback import CallbackBase
-
-DOCUMENTATION = '''
-    callback: log_to_file
-    type: notification
-    short_description: output logs to a file
-    description:
-        - This summarises tasks per host and outputs it to a log file
-'''
+import os
 
 class CallbackModule(CallbackBase):
-    """
-    logs playbook results, per host, in /var/log/ansible/hosts
-    """
+
     CALLBACK_VERSION = 2.0
     CALLBACK_TYPE = 'notification'
     CALLBACK_NAME = 'log_to_file'
     CALLBACK_NEEDS_WHITELIST = True
 
-    MSG_FORMAT = "{task}={status}\n"
-
     def __init__(self):
-
         super(CallbackModule, self).__init__()
-        self.current_task = ''
-        self.passed = 0
-        self.failed = 0
         self.output_dir = os.getcwd()
-        self.path = os.path.join(self.output_dir, "test_run_result.out")
-        # either create the file or truncate the existing file
-        with open(self.path, 'w') as fp:
-            pass
-
-    def playbook_on_task_start(self, task, is_conditional):
-        self.current_task = str(task)
-
-#    def playbook_on_stats(self, stats):
-#        with open(self.path, 'a') as fd:
-#            fd.write("passed: {}\n".format(self.passed))
-#            fd.write("failed: {}\n".format(self.failed))
-
-    def log(self, host, category):
-         # only interested in the test tasks, not setup or debug
-         if self.current_task.startswith("RHELOSP"):
-
-             if category == "failed":
-                self.failed +=1
-             if category == "passed":
-                 self.passed +=1
-
-             test_result_message = self.MSG_FORMAT.format(task=self.current_task, status=category)
```

Comment:
There's a requirement on the format of the results.out file to remain consistent.
The format needs to be "{task_name}={passed|failed}"

Reply:
The format should be {testID}={passed|failed}

Existing users expect the test_run_result.out file to contain results in this format.

Reply:
I kept the file "test_run_result.out" using the same old format , I just updated the functions to be more organised and we will have another file called "{host}_results.log" that will have more details so we can use it in the zuul job.